client/components/payment-activity/date-range-picker.tsx

```diff
@@ -0,0 +1,148 @@
+	.clone()
+	.set( { hour: 23, minute: 59, second: 59, millisecond: 0 } );
+const timeOptions: {
+	[ key: string ]: {
+		start: moment.Moment;
+		end: moment.Moment;
+		displayKey: string;
+	};
+} = {
+	today: {
+		start: now
+			.clone()
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: todayEndOfDay,
+		displayKey: __( 'Today', 'woocommerce-payments' ),
+	},
+	last_7_days: {
+		start: now
+			.clone()
+			.subtract( 7, 'days' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 7 days', 'woocommerce-payments' ),
+	},
+	last_4_weeks: {
+		start: now
+			.clone()
+			.subtract( 4, 'weeks' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 4 weeks', 'woocommerce-payments' ),
+	},
+	last_3_months: {
+		start: now
+			.clone()
+			.subtract( 3, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 3 months', 'woocommerce-payments' ),
+	},
+	last_12_months: {
+		start: now
+			.clone()
+			.subtract( 12, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 12 months', 'woocommerce-payments' ),
+	},
+	month_to_date: {
+		start: now.clone().startOf( 'month' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Month to date', 'woocommerce-payments' ),
+	},
+	quarter_to_date: {
+		start: now.clone().startOf( 'quarter' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Quarter to date', 'woocommerce-payments' ),
+	},
+	year_to_date: {
+		start: now.clone().startOf( 'year' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Year to date', 'woocommerce-payments' ),
+	},
+};
+
+const formatDateRange = (
+	start: moment.Moment,
+	end: moment.Moment
+): string => {
+	// Today - show only today's date.
+	if ( start.isSame( end, 'day' ) ) {
+		return start.format( 'MMMM D, YYYY' );
+	}
+
+	// Different years - show year for both start and end
+	if ( ! start.isSame( end, 'year' ) ) {
+		return `${ start.format( 'MMMM D, YYYY' ) } - ${ end.format(
+			'MMMM D, YYYY'
+		) }`;
+	}
+
+	// Same year - show year only for end date.
+	return `${ start.format( 'MMMM D' ) } - ${ end.format( 'MMMM D, YYYY' ) }`;
+};
+
+const options = Object.keys( timeOptions ).map( ( optionKey ) => {
+	const option = timeOptions[ optionKey ];
+	return {
+		key: optionKey,
+		name: option.displayKey,
+		hint: formatDateRange( option.start, option.end ),
+	};
+} );
+
+export const defaultDateRange = {
+	date_start: timeOptions.last_7_days.start.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+	date_end: timeOptions.last_7_days.end.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+};
+
+export const DateRangePicker: React.FC< {
+	onDateRangeChange: ( dateRange: DateRange ) => void;
+} > = ( { onDateRangeChange } ) => {
+	const [ selectedOption, setSelectedOption ] = useState(
+		options.find( ( option ) => option.key === 'last_7_days' )
+	);
```

Comment:
I've noticed a bug with the selected option reverting to the default `last_7_days` (which may have been introduced by the `/data` dir changes I pushed).

https://github.com/Automattic/woocommerce-payments/assets/3147296/4b47c6d9-1832-44f0-8f1f-79aa6349c4c8


It looks like the default selected state here is reverting to `last_7_days` when the parent component re-renders – when a new date is selected.

Rather than have the "selected date" state within this component, we could treat the date range query param as the source of state and pass this to the `DateRangePicker`.

https://github.com/Automattic/woocommerce-payments/blob/1a0c23ac2c27d6d539bf54fabefff3ade6d8c006/client/components/payment-activity/index.tsx#L58-L61

The `DateRangePicker` could then infer the selected date option based on the passed option.

Reply:
Thanks @Jinksi for the pointer
Fixed in [8c65863](https://github.com/Automattic/woocommerce-payments/pull/8927/commits/8c65863498111eb0540874853c6eff1a78095887)